app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -196,52 +236,74 @@ const buildTree = (
     (!previousElementNode.isArray || rootNode === previousElementNode)
   ) {
     previousElementNode.children.push(currentElementNode);
-    buildTree(rest, rootNode, currentElementNode);
+    buildTree(rest, rootNode, currentElementNode, parentPath);
   } else {
     const parent = getParent(previousElementNode, rootNode);
-    if (parent) buildTree(elementsDefinition, rootNode, parent);
+    if (parent) buildTree(elementsDefinition, rootNode, parent, parentPath);
   }
 };
 
 const useFhirResourceTreeData = (
   params: {
     id: string;
     nodeId?: string;
+    nodePath?: string;
   },
   options?: { skip?: boolean }
 ): {
   root?: ElementNode;
-  isLoading: boolean;
+  isStructureDefinitionLoading: boolean;
+  isAttributesLoading: boolean;
 } => {
-  const { id, nodeId } = params;
+  const { id, nodeId, nodePath } = params;
   const {
     data: structureDefinition,
-    isLoading,
+    isLoading: isStructureDefinitionLoading,
   } = useApiStructureDefinitionRetrieveQuery(
     {
       id,
     },
     options
   );
+  const { mappingId } = useParams<{ mappingId?: string }>();
   const dispatch = useAppDispatch();
   const root = useAppSelector(selectRoot);
+  const {
+    data: attributes,
+    isLoading: isAttributesLoading,
+  } = useApiAttributesListQuery({ resource: mappingId });
 
   const data = useMemo(() => {
     if (structureDefinition?.snapshot) {
       const elementDefinitions = structureDefinition.snapshot.element;
-      const rootNode = createElementNode(elementDefinitions[0]);
-      buildTree(elementDefinitions.slice(1), rootNode, rootNode);
+      const rootNode = createElementNode(elementDefinitions[0], {
+        parentPath: nodePath,
+      });
+      buildTree(elementDefinitions.slice(1), rootNode, rootNode, nodePath);
       return rootNode;
     }
-  }, [structureDefinition]);
+  }, [structureDefinition, nodePath]);
+
+  const dataAttributes = useMemo(() => {
+    if (attributes) {
+      const elementNodes: ElementNode[] = [];
+      attributes.forEach((attribute) => {
+        const elementDefinition = createElementDefinition(attribute);
+        const newElementNode = createElementNode(elementDefinition, {});
+        newElementNode.definition.path = newElementNode.definition.id;
+        elementNodes.push(newElementNode);
+      });
+      return elementNodes;
+    }
+  }, [attributes]);
```

Comment:
It would be better if this were merged with the other `useMemo`

Let's say you generate an array of `ElementDefinition` from the array of `Attribute`, then you could just concatenate it to the array of `ElementDefinition`extracted from the StructureDefinition's snapshot and just let the tree algorithm do the rest ?

This seems like it could work, but some changes may be done within the `buildTree` method still